Add explanation of IDE design process and include template (#61751)
dotnet/roslyn: .github/ISSUE_TEMPLATE/feature-suggestion.md

---

> name: Feature suggestion

Reviewer: Should this say "IDE Feature suggestion", with a note about putting language feature suggestions in csharplang?

about: Propose a new feature
title: ''
labels: [Feature Request, Area-IDE]
assignees: ''
---

<!--

> We welcome feature suggestions! We have a process to discuss and evaluate new feature suggestions. Here is an overview of our process [here](https://github.com/dotnet/roslyn/blob/main/docs/contributing/ide_design_process.md). This template will help us gather the information we eed to start the review process.

Reviewer 2: ```suggestion
We welcome feature suggestions! We have a process to discuss and evaluate new feature suggestions. Here is an overview of our process [here](https://github.com/dotnet/roslyn/blob/main/docs/contributing/ide_design_process.md). This template will help us gather the information we need to start the review process.
```

-->

## Summary

<!--
Brief summary of what this proposal is about.
-->

## Background and Motivation

<!--

> What the is problem we are solving and in what context did you encounter it?

Reviewer 2: ```suggestion
What is the problem we are solving and in what context did you encounter it?
```

-->

## Proposed Feature

<!--

> Please provide a sketch of the feature you are proposing. Be as specific as you can: the more pecific the proposal, the easier the process will be. Including screenshots of some of the xisting problems can also help a lot here.

Reviewer 2: ```suggestion
Please provide a sketch of the feature you are proposing. Be as specific as you can: the more specific the proposal, the easier the process will be. Including screenshots of some of the existing problems can also help a lot here.
```
